### PR TITLE
spawner.py update for default shell

### PR DESCRIPTION
**Pull request by jdavidheiser**
Update spawner.py to do a better job keeping the shell variable intact for terminals launched from within the notebook.

Exporting the SHELL environment variable when launching jupyterhub should now use that shell for terminado terminals.  I set this in my /etc/init.d/jupyter script.  This should fix issue #133

### CODE REVIEW

File: jupyterhub/spawner.py [hunk 1]
```diff
@@ -81,6 +81,7 @@ class Spawner(LoggingConfigurable):
         'VIRTUAL_ENV',
         'LANG',
         'LC_ALL',
+        'SHELL'
```
Comment (minrk) [inline jupyterhub/spawner.py]: The SHELL of the server terminal (often sh) user shouldn't set the SHELL of the single-user servers.

**Comment by minrk** [comment]
I think the right fix is to set the shell for the user with `pwd.getpwnam(User.name).pw_shell` in LocalProcessSpawner.

**Comment by jdavidheiser** [comment]
I was mostly trying to address the issue brought up in #133, by supporting the "set shell variable then launch jupyterhub" use, but I agree that using the user-specific shell is much cleaner.  I'll try to test the change later this afternoon and submit an updated PR.

**Comment by jdavidheiser** [comment]
the update as suggested worked as expected, and user specified shells are respected.

**Comment by minrk** [comment]
Thanks, merged.

**Pull request merged**